cli/src/commands/config.rs

```diff
@@ -78,8 +78,26 @@ pub(crate) struct ConfigListArgs {
     /// Whether to explicitly include built-in default values in the list.
     #[arg(long)]
     pub include_defaults: bool,
+    /// Target the user-level config
+    #[arg(long)]
+    user: bool,
+    /// Target the repo-level config
+    #[arg(long)]
+    repo: bool,
     // TODO(#1047): Support --show-origin using LayeredConfigs.
-    // TODO(#1047): Support ConfigArgs (--user or --repo).
+}
+
+impl ConfigListArgs {
+    fn get_source_kind(&self) -> Option<ConfigSource> {
+        if self.user {
+            Some(ConfigSource::User)
+        } else if self.repo {
+            Some(ConfigSource::Repo)
```

Comment:
Since these values are mutually exclusive here (which seems reasonable), we should also tell clap that they are mutually exclusive so the user will get an error if they try `jj config list --user --repo`. You should be able to add `conflicts_with = "user"` in the attribute on `repo`.